File: py/IEC61968/DER/DispatchablePowerCapability.py
# Converted by an OPENAI API call using model: gpt-3.5-turbo-1106
from CIM_STD_PYTHON.TC57CIM.IEC61970.Base.Domain.ActivePower import ActivePower
from CIM_STD_PYTHON.TC57CIM.IEC61970.Base.Domain.ApparentPower import ApparentPower
from CIM_STD_PYTHON.TC57CIM.IEC61970.Base.Domain.ReactivePower import ReactivePower


class DispatchablePowerCapability:
    
    def __init__(self):
        self.current_active_power = ActivePower()
        self.current_apparent_power = ApparentPower()
        self.current_reactive_power = ReactivePower()
        self.max_active_power = ActivePower()
        self.max_apparent_power = ApparentPower()
        self.max_reactive_power = ReactivePower()
        self.min_active_power = ActivePower()
        self.min_apparent_power = ApparentPower()
        self.min_reactive_power = ReactivePower()

File: java/IEC61968/DER/DispatchablePowerCapability.java
package IEC61968.DER;

import IEC61970.Base.Domain.ActivePower;
import IEC61970.Base.Domain.ApparentPower;
import IEC61970.Base.Domain.ReactivePower;

/**
 * @author marga
 * @version 1.0
 * @created 25-Dec-2023 8:45:20 PM
 */
public class DispatchablePowerCapability {

	/**
	 * Product of RMS value of the voltage and the RMS value of the in-phase component
	 * of the current
	 */
	public ActivePower currentActivePower;
	/**
	 * Product of the RMS value of the voltage and the RMS value of the current
	 */
	public ApparentPower currentApparentPower;
	/**
	 * Product of RMS value of the voltage and the RMS value of the quadrature
	 * component of the current
	 */
	public ReactivePower currentReactivePower;
	/**
	 * Product of RMS value of the voltage and the RMS value of the in-phase component
	 * of the current
	 */
	public ActivePower maxActivePower;
	/**
	 * Product of the RMS value of the voltage and the RMS value of the current
	 */
	public ApparentPower maxApparentPower;
	/**
	 * Product of RMS value of the voltage and the RMS value of the quadrature
	 * component of the current
	 */
	public ReactivePower maxReactivePower;
	/**
	 * Product of RMS value of the voltage and the RMS value of the in-phase component
	 * of the current
	 */
	public ActivePower minActivePower;
	/**
	 * Product of the RMS value of the voltage and the RMS value of the current
	 */
	public ApparentPower minApparentPower;
	/**
	 * Product of RMS value of the voltage and the RMS value of the quadrature
	 * component of the current
	 */
	public ReactivePower minReactivePower;

	public DispatchablePowerCapability(){

	}
}//end DispatchablePowerCapability